anomalib/utils/cli/cli.py

```diff
@@ -0,0 +1,239 @@
+
+        Args:
+            parser (LightningArgumentParser): Lightning Argument Parser.
+        """
+        # TODO: https://github.com/openvinotoolkit/anomalib/issues/19
+        # TODO: https://github.com/openvinotoolkit/anomalib/issues/20
+        parser.add_argument("--openvino", type=bool, default=False, help="Export to ONNX and OpenVINO IR format.")
+        parser.add_argument("--nncf", type=str, help="Path to NNCF config to enable quantized training.")
+
+        # ADD CUSTOM CALLBACKS TO CONFIG
+        # NOTE: MyPy gives the following error:
+        # Argument 1 to "add_lightning_class_args" of "LightningArgumentParser"
+        # has incompatible type "Type[TilerCallback]"; expected "Union[Type[Trainer],
+        # Type[LightningModule], Type[LightningDataModule]]"  [arg-type]
+        parser.add_lightning_class_args(TilerConfigurationCallback, "tiling")  # type: ignore
+        parser.set_defaults({"tiling.enable": False})
+
+        parser.add_lightning_class_args(MetricsConfigurationCallback, "metrics")  # type: ignore
+        parser.set_defaults(
+            {
+                "metrics.adaptive_threshold": True,
+                "metrics.default_image_threshold": None,
+                "metrics.default_pixel_threshold": None,
+                "metrics.image_metric_names": ["F1Score", "AUROC"],
+                "metrics.pixel_metric_names": ["F1Score", "AUROC"],
+                "metrics.normalization_method": "min_max",
+            }
+        )
+
+        parser.add_lightning_class_args(VisualizerCallback, "visualization")  # type: ignore
+        parser.set_defaults(
+            {
+                "visualization.task": "segmentation",
+                "visualization.log_images_to": ["local"],
+                "visualization.inputs_are_normalized": True,
+            }
+        )
+
+    def before_instantiate_classes(self) -> None:
+        """Modify the configuration to properly instantiate classes."""
+
+        # Get the root dir.
+        # NOTE: This always assumes that the script calls 'fit' subcommand. This may not always be the case.
+
+        # subcommand could be <fit, test, predict or tune>.
+        subcommand = self.config["subcommand"]
+        # Configurations are stored in self.config.<fit,test,predict,tune>
+        config = self.config[subcommand]
+
+        # If `resume_from_checkpoint` is not specified, it means that the project has not been created before.
+        # Therefore, we need to create the project directory first.
+        if config.trainer.resume_from_checkpoint is None:
+            root_dir = config.trainer.default_root_dir if config.trainer.default_root_dir else "./results"
+            model_name = config.model.class_path.split(".")[-1].lower()
+            data_name = config.data.class_path.split(".")[-1].lower()
+            category = config.data.init_args.category if config.data.init_args.keys() else ""
+            time_stamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
```

Comment:
What do you think about this `time_stamp` here?

Reply:
I don't see an issue with it.